src/server_manager/web_app/app.ts

```diff
@@ -920,24 +956,107 @@ export class App {
         });
   }
 
+  private async setAccessKeyDataLimit(limit: server.DataLimit) {
+    if (!limit) {
+      return;
+    }
+    const previousLimit = this.selectedServer.getAccessKeyDataLimit();
+    if (previousLimit && limit.bytes === previousLimit.bytes) {
+      return;
+    }
+    const serverView = this.appRoot.getServerView(this.appRoot.selectedServer.id);
+    try {
+      await this.selectedServer.setAccessKeyDataLimit(limit);
+      this.appRoot.showNotification(this.appRoot.localize('saved'));
+      serverView.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(limit);
+      this.refreshTransferStats(this.selectedServer, serverView);
+    } catch (error) {
+      console.error(`Failed to set access key data limit: ${error}`);
+      this.appRoot.showError(this.appRoot.localize('error-set-data-limit'));
+      serverView.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(
+          previousLimit || await this.computeDefaultAccessKeyDataLimit(this.selectedServer));
+      serverView.isAccessKeyDataLimitEnabled = !!previousLimit;
+    }
+  }
+
+  private async removeAccessKeyDataLimit() {
+    const serverView = this.appRoot.getServerView(this.appRoot.selectedServer.id);
+    try {
+      await this.selectedServer.removeAccessKeyDataLimit();
+      this.appRoot.showNotification(this.appRoot.localize('saved'));
+      this.refreshTransferStats(this.selectedServer, serverView);
+    } catch (error) {
+      console.error(`Failed to remove access key data limit: ${error}`);
+      this.appRoot.showError(this.appRoot.localize('error-remove-data-limit'));
+      serverView.isAccessKeyDataLimitEnabled = true;
+    }
+  }
+
+  private dataLimitToDisplayDataAmount(limit: server.DataLimit): DisplayDataAmount|null {
+    if (!limit) {
+      return null;
+    }
+    const bytes = limit.bytes;
+    if (bytes >= 10 ** 9) {
+      return {value: Math.round(bytes / (10 ** 9)), unit: 'GB'};
```

Comment:
Also worth noting that due to the rounding that these two conversion functions don't round-trip.  Maybe it's possible instead to do a string conversion and truncate to a maximum length?

Reply:
I mean as in having the ability to display "1.5 GB" seems reasonable, but we'd have to deal with repeating fractions